<launch>

    <arg name="camera1" default="laptop_cam" />
    <arg name="camera2" default="giacomo" />
    

    <node name="$(arg camera1)" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device" value="/dev/video0" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_name" value="laptop_cam" />
    </node>

    <node name="$(arg camera2)" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device" value="/dev/video1" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_name" value="giacomo" />
        <param name="camera_frame_id" value="usb_cam" />
    </node>

    <node name="ball_position" pkg="stewart_platform" type="ball_tracker2.py" output="screen">
        <param name="camera_name" value="$(arg camera1)" />
    </node>

    <node name="ball_position2" pkg="stewart_platform" type="ball_tracker2.py" output="screen">
        <param name="camera_name" value="$(arg camera2)" />
        <remap from="ball_position" to="ball_position2" />
    </node>

    <node name="dual_cam" pkg="stewart_platform" type="dual_cam.py" output="screen"/>





    <param name="robot_description" command="$(find xacro)/xacro.py $(find stewart_platform)/urdf/usb_cam.urdf"/>
    <node pkg="robot_state_publisher" type="state_publisher" name="state_publisher"/>

    




    <arg name="marker_size" default="4.4" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />

    <arg name="output_frame" default="/usb_cam" />
    <arg name="cam_image_topic" default="/giacomo/image_raw" />
    <arg name="cam_info_topic" default="/giacomo/camera_info" />

    <arg name="bundle_files" default="$(find ar_track_alvar)/bundles/truthTableLeg.xml $(find ar_track_alvar)/bundles/table_8_9_10.xml" />



    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="maker_size" type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error" type="double" value="$(arg max_track_error)" />
        <param name="output_frame" type="string" value="$(arg output_frame)" />

        <remap from="camera_image" to="$(arg cam_image_topic)" />
        <remap from="camera_info" to="$(arg cam_info_topic)" />
    </node>



</launch>
